label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRule data = 2;
+}
+
+message CreateLabelApplicationRule {
+  LabelApplicationRule data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  string description = 2;
+  MatchingCriteria matching_criteria = 3;
+  ApplicationCriteria application_criteria = 4;
+
+  message MatchingCriteria {
+    Type type = 1;
+    repeated Condition conditions = 2;
+
+    enum Type {
+      TYPE_UNSPECIFIED = 0;
+      TYPE_SPAN = 1;
+      TYPE_TRACE = 2;
+    }
+
+    message Condition {
+      string name = 1;
+      Scope scope = 2;
+      string key = 3;
+      Operator operator = 4;
+      optional string value = 5;
+
+      enum Scope {
+        SCOPE_UNSPECIFIED = 0;
+        SCOPE_ATTRIBUTE = 1;
+        SCOPE_REQUEST = 2;
```

Comment:
request and response are just attributes (well, requestBody/requestHeaders/responseBody/responseHeaders are). Do we need to separate them? This also differs from how we use "scope" elsewhere in the platform and in this PR.

Reply:
Even though your point is valid, not user whether end user will prefer that distinction. A single bucket of attributes is simpler in terms of implementation. Will check with Sudeep on this front.

Reply:
Yes there's the end user question, but that's not necessarily something we need to capture at this level. First and foremost, we want this to be flexible, so we need not build out special cases for ease of use - we'd rather be generic and handle the ease of use at a higher layer.

Reply:
If we don't capture user's input as is in config, then we will need to transform both ways - from user input to what's persisted in config, as well as config to what's presented to user.

Reply:
We'll have to transform either way, since we're not presenting protos to users :)